dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/SupervisionOrderRestRepository.java

```diff
@@ -0,0 +1,239 @@
+public class SupervisionOrderRestRepository extends DSpaceRestRepository<SupervisionOrderRest, Integer> {
+
+    private static final Logger log =
+        org.apache.logging.log4j.LogManager.getLogger(SupervisionOrderRestRepository.class);
+
+    @Autowired
+    private SupervisionOrderService supervisionOrderService;
+
+    @Autowired
+    private ConverterService converterService;
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private GroupService groupService;
+
+    @Autowired
+    private AuthorizeService authorizeService;
+
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @Override
+    public SupervisionOrderRest findOne(Context context, Integer id) {
+        try {
+            SupervisionOrder supervisionOrder = supervisionOrderService.find(context, id);
+            if (Objects.isNull(supervisionOrder)) {
+                throw new ResourceNotFoundException("Couldn't find supervision order for id: " + id);
+            }
+            return converterService.toRest(supervisionOrder, utils.obtainProjection());
+        } catch (SQLException e) {
+            log.error("Something went wrong with getting supervision order with id:" + id, e);
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @Override
+    public Page<SupervisionOrderRest> findAll(Context context, Pageable pageable) {
+        try {
+            List<SupervisionOrder> supervisionOrders = supervisionOrderService.findAll(context);
+            return converterService.toRestPage(supervisionOrders, pageable, utils.obtainProjection());
+        } catch (SQLException e) {
+            log.error("Something went wrong with getting supervision orders", e);
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @Override
+    public SupervisionOrderRest createAndReturn(Context context) throws AuthorizeException, SQLException {
+        HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
+        SupervisionOrder supervisionOrder;
+        String itemId = req.getParameter("uuid");
+        String groupId = req.getParameter("group");
+        String type = req.getParameter("type");
+
+        validateParameters(itemId, groupId, type);
+
+        Item item = itemService.find(context, UUID.fromString(itemId));
+        if (item == null) {
+            throw new UnprocessableEntityException("Item with uuid: " + itemId + " not found");
+        }
+
+        Group group = groupService.find(context, UUID.fromString(groupId));
+        if (group == null) {
+            throw new UnprocessableEntityException("Group with uuid: " + groupId + " not found");
+        }
+
+        supervisionOrder = supervisionOrderService.findByItemAndGroup(context, item, group);
+        if (Objects.nonNull(supervisionOrder)) {
+            throw new ResourceConflictException(
```

Comment:
by previous agreement this special kind of conflict related to business rule was decided to be managed with a 422 see for instance here https://github.com/DSpace/RestContract/pull/180#discussion_r850829376